projects/step-core/src/lib/modules/table/services/items-per-page.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { UserService } from '../../../client/generated';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ItemsPerPageService {
+  private defaultItemsPerPage: Array<number> = [10, 25, 50, 100];
+  private customItemsPerPage?: number | boolean;
+
+  constructor(private _userService: UserService) {}
+
+  public getItemsPerPage(loadedUserPreferences: Function): any {
+    if (this.customItemsPerPage === undefined) {
+      this._userService.getPreferences().subscribe((preferences) => {
+        if (preferences?.preferences?.['tables_itemsperpage']) {
+          this.customItemsPerPage = Number(preferences.preferences['tables_itemsperpage']);
```

Comment:
the expression ```Number(preferences.preferences['tables_itemsperpage'])``` exists twice, it would be nice to have it as a constant

Reply:
```suggestion
          this.customItemsPerPage = Number(tables_itemsperpage);
```

Reply:
```suggestion
          this.customItemsPerPage = Number(preferences.preferences['tables_itemsperpage']);
```